### PR TITLE
Add find_all_by_city method to do reverse lookup on IATA

### PR DESCRIPTION
**Pull request by viral810**
This method is useful in any case where you would want to grab `iata` code by the city name

would return `nil` if no airport found and an `array` of airports if found.  It should be up to users to whatever they want to do with the returned data.

In my case, I filter out returned list of airports and if no international airport found then typically it would be only one airport in that city.

### CODE REVIEW

File: lib/airports.rb [hunk 1]
```diff
@@ -12,6 +12,19 @@ def self.find_by_iata_code(iata_code)
     Airport.
       new(airport_data.each_with_object({}) { |(k, v), h| h[k.to_sym] = v })
   end
+  
+  def self.find_by_city(city)
+    return unless city.length < 1
```
Comment (timrogers) [inline lib/airports.rb]: I'd ✂️ this - not really any need for it I think.

File: lib/airports.rb [hunk 2]
```diff
@@ -12,6 +12,19 @@ def self.find_by_iata_code(iata_code)
     Airport.
       new(airport_data.each_with_object({}) { |(k, v), h| h[k.to_sym] = v })
   end
+  
+  def self.find_by_city(city)
```
Comment (timrogers) [inline lib/airports.rb]: Is this method useful? What about where a city has multiple airports? Would you want the one that *just so happens* to be first in the dataset?

File: lib/airports.rb (hunk 1 above)
Comment (viral810) [reply lib/airports.rb]: make sense

File: lib/airports.rb [hunk 3]
```diff
@@ -12,6 +12,10 @@ def self.find_by_iata_code(iata_code)
     Airport.
       new(airport_data.each_with_object({}) { |(k, v), h| h[k.to_sym] = v })
   end
+  
+  def self.find_all_by_city(city)
```
Comment (timrogers) [inline lib/airports.rb]: Can we rename this to `by_city`?  That feels slightly nicer. It'd also be good to have some tests.

**Comment by grammakov** [comment]
Would love to see this go live!

**Comment by timrogers** [comment]
@grammakov Feel free to pick up this PR!

**Comment by viral810** [comment]
@timrogers Let me know if this hasn't gone live yet, would love to pick this up back and get it done

**Comment by timrogers** [comment]
@viral810 Happy to merge if the comments above are addressed.

**Pull request closed without merging**